01-30-shield-blocking/actors/player/Player.gd

```diff
@@ -0,0 +1,78 @@
+extends KinematicBody2D
+class_name Player
+
+onready var invencible_timer : Timer = $InvencibleTimer
+onready var animated_sprite : AnimatedSprite = $AnimatedSprite
+onready var animation_player : AnimationPlayer = $AnimationPlayer
+onready var shield : Node2D = $Shield
+
+export var move_speed : = 150.0
+export var jump_force : = 200.0
+export var gravity : = 25.0
+
+var velocity : = Vector2()
+var blocking : = false
+
+
+func _process(delta: float) -> void:
+	if Input.is_action_just_pressed("defend"):
+		shield.appear(not animated_sprite.flip_h)
```

Comment:
If the method is "appear", you shouldn't pass a boolean to it. Appear suggests that this method should make the shield visible. You could use something like `shield.visible = bool` instead

Reply:
> Appear suggests that this method should make the shield visible. 

But that's exactly what it does, and fixes its position + enables collision. I'll connect to draw and hide and check for inputs in the shield to know when to flip its position then.